tensorflow/python/keras/distribute/distributed_training_utils.py

```diff
@@ -645,6 +645,8 @@ def _prepare_feed_values(model, inputs, targets, sample_weights, mode):
     targets = []
   elif not is_distributing_by_cloning(model):
     sample_weights = None  # b/129503665
```

Comment:
Hey @a6802739!  Thanks for making this change, it's important for us.

I was passing by and noticed that you were trying to make the eager case work.  I think in Eager `sample_weights` at this point should `PerReplica` objects.  Right now they are overwritten with `None`.  I think instead those PerReplica objects needs to be directly passed into `ins` without being overwritten.   That's what it seems to me.  I hope this helps and good luck!